### ISSUE TITLE
Custom shortcut tags and fuzzy search

### ISSUE THREAD

**Issue by fehC** [BranchMetrics/sesame_issues#43]
Hello,

Could you please try to implement the following functions:

**Custom Tags:**

An option to add custom shortcut tags.
_Example:_
XiiaLive app, I would like to add tags like "music, radio, media etc", so typing any of the tag words would include XiiaLive in the find list.

**Fuzzy Search:**

An option to add fuzzy search function.
_Example:_
Typing "le" would find/list "Gal**le**ry" and "Goog**le**" and any shortcut that contains "le" anywhere in it's name/tags.

I'm willing to donate extra money for this, just provide a donation link if you're willing to work on it.

Thanks

**Comment by Chaphasilor** [comment]
Yes, the fuzzy search aspect also bugs me sometimes. I use sesame as my launcher, and I don't have an app drawer, so if I don't know the name of an app it's a real struggle. But sometimes I know part of the name, sesame still won't find it though...

**Comment by philwall2** [comment]
Good ideas.

Re: tags, we plan to let you label shortcuts, see #14 . I see how tags would be useful, the problem is we want to keep the search UI super clean. One thing that's different about Sesame vs the Google App's search is we don't put categories, multiple rows of text etc. We think (hope) that makes it overall easier to use because it's less cognitive load.

I get that adding a label to every music shortcut in XiiaLive would be annoying, but there's other ways Sesame ranks things: it remembers prior choices, it also matches first letters and ranks those up. e.g. if I want to access "XiiaLive: Gorillaz", I could type "X, G", and it will be up there. If I use it, then next time it will be #1, and even just typing "X" might work.

Our theory is if we keep the results UI perfectly clean, you can use the first letters + history ranking to get to anything in 1 or 2 taps.

What do you think?
---------
Fuzzy search option is a good idea. We can do that.

**Comment by fehC** [comment]
@philwall2 
Tagging items and adding fuzzy search as **options** wouldn't clutter the search UI. You could even add them off by default.

I have been using FARR (Find and Run Robot) on windows for years, can't imagine using the computer without it.
In FARR you could custom categories aliases. I have app category that gets triggered when I type "app " app+space. Example:
If I type "app sh" FARR would list all  entries in app category that contains "sh" anywhere in its name or tags.

Adding these two options to Sesame would make my digital life perfect.

Please consider it.
Thank you

**Comment by verboze** [comment]
I really look forward to tags. I mainly want this as a way to group apps and to remember apps I don't use often. For example, I have travel apps I seldom use, but I can't remember them by name. Being able to type "travel" to list those apps would be swell. Bonus: allow explicit search by tag via a hash prefix to the tag, and allow search by multiple tags; so I would search for "#travel, #hotel" to list all apps tagged travel or hotel.

**Comment by fehC** [comment]
Still waiting for these 2 features to be added. Starting labels/tags with # should be optional. I wouldn't want to start searching for custom tags starting with # since that means I have to go to another page on my keyboard. The whole idea behind this app is speed.
Come on developers, get it done :)
And please don't forget fuzzy search.

**Comment by Fred-Vatin** [comment]
Fuzzy search 🙏🏻

That keeps me out of using sesame most of the time because I can not just find what I search.

When you're accustomed to fuzzy search, it's hard to not use it.

**Comment by whitenoisedev** [comment]
Hi perhaps this FR should be split in two for better tracking.

> it remembers prior choices, it also matches first letters and ranks those up. e.g. if I want to access "XiiaLive: Gorillaz", I could type "X, G", and it will be up there. If I use it, then next time it will be #1, and even just typing "X" might work.

@Chaphasilor 
This first letter matching does not work for me.
